types/async/index.d.ts

```diff
@@ -15,7 +15,7 @@
 export as namespace async;
 
 export interface Dictionary<T> { [key: string]: T; }
-export type IterableCollection<T> = T[] | IterableIterator<T> | AsyncIterable<T> | Dictionary<T>;
+export type IterableCollection<T> = T[] | Iterable<T> | AsyncIterable<T> | Dictionary<T>;
```

Comment:
That's a further deviation from the official docs though. And in that case it would make sense to add AsyncIterableIterator too, which is a further deviation.